[tflite] Support tflite v2.7.0 for tizen and debian

anyj0527 opened the pull request:
- Remove unnecessary gles dependency
- Support newly introduced xnnpack delegate
- Remove unnecessary build options
- Set tensor_filter_ext_enabled when tflite2_support is available
- Add a gtest based test to examine the XNNPACK delegate

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

myungjoo commented on ext/nnstreamer/tensor_filter/meson.build:
```diff
@@ -219,6 +219,10 @@ if tflite2_support_is_available
+  if cc.has_header('tensorflow2/lite/delegates/xnnpack/xnnpack_delegate.h') or cc.has_header('tensorflow/lite/delegates/xnnpack/xnnpack_delegate.h')
+    tflite2_compile_args += '-DTFLITE_XNNPACK_DELEGATE_SUPPORTED'
+  endif
+
```
Let's define ```tflite2-xnnpack-delegate-support``` with ```type: 'feature'``` and run 222-224 according to the given options (auto, disabled, enabled)

myungjoo replied:
This is to provide controllability for platform build and integration.

anyj0527 replied:
As Tizen would use tf2lite which supports XNNPACK, I set the option as `true` and let the meson check the header for it during the build configuration. PTAL

myungjoo replied:
with type:'feature'

if value=enabled
   error if header does not exist
   enable if exist
elseif value=auto
   enable if exist
   disable if not exist
else
   disable

dongju-chae reviewed (approved):
LGTM

gichan-jang commented on tests/nnstreamer_filter_tensorflow2_lite/unittest_filter_tensorflow2_lite.cc:
```diff
@@ -233,6 +233,45 @@ TEST (nnstreamerFilterTensorFlow2Lite, floatModelResult)
+  g_usleep (1000 * 1000 * 5); // wait for 5 seconds to check all output is valid
+
+  EXPECT_EQ (setPipelineStateSync (gstpipe, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+
```
unref `sink_handle` here?

anyj0527 replied:
Thanks! I added the line.

again4you commented on tests/nnstreamer_filter_tensorflow2_lite/unittest_filter_tensorflow2_lite.cc:
```diff
@@ -233,6 +233,45 @@ TEST (nnstreamerFilterTensorFlow2Lite, floatModelResult)
+ */
+TEST (nnstreamerFilterTensorFlow2Lite, floatModelXNNPACKResult)
+{
+  gchar *pipeline;
```
In my humble opinion, `g_autofree` is useful for this case.
```cpp
g_autofree gchar *pipeline = g_strdup_printf(...);
```

anyj0527 replied:
Well I think manually handling the string seems ok.. Can you explain what is the pros on this case?

myungjoo commented on ext/nnstreamer/tensor_filter/meson.build:
```diff
@@ -239,7 +239,12 @@ if tflite2_support_is_available
+    if cc.has_header('tensorflow2/lite/delegates/xnnpack/xnnpack_delegate.h') or cc.has_header('tensorflow/lite/delegates/xnnpack/xnnpack_delegate.h')
+      tflite2_compile_args += '-DTFLITE_XNNPACK_DELEGATE_SUPPORTED'
+    else
+      warning('There is no header file for xnnpack delegate. Disable it')
```
If ```tflite2-xnnpack-delegate-support``` is true, this should be error, not warning.
That's why I'm telling you to use type:'feature' instead of type:'boolean' for your needs.

Anyway, 
```
option('tflite2-xnnpack-delegate-support', type: 'boolean', value: 'true')
```
is strange too. If what you want to is "auto", use 'feature".

anyj0527 commented:
@jaeyun-jung @myungjoo I updated this PR after the tflite v2.7.0 tizen and debian packages. Removed unnecessary build options and declaration.
It can be merged after the tflite v2.7.0 is released for all tizen profiles.

jaeyun-jung commented on packaging/nnstreamer.spec:
```diff
@@ -195,9 +194,6 @@ BuildRequires: tensorflow-lite-devel
 %if 0%{?tensorflow2_lite_support}
 # for tensorflow2-lite
 BuildRequires: tensorflow2-lite-devel
-%if 0%{?tensorflow2_gpu_delegate_support}
```
consider to change this requirements for the test
```
%if 0%{?tensorflow2_lite_support}
BuildRequires: pkgconfig(gles20)
%endif
```

anyj0527 replied:
Currently, the filter does not support gl backend (only opencl backend for gpu delegate). I think gles20 dep is not required at this time.

myungjoo reviewed (approved):
LGTM

anyj0527 commented:
Marking this PR as ready to go as TFLite v2.7.0 was released for all supported profiles.
